sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3104,6 +3104,22 @@ object GpuOverrides extends Logging {
         override val supportOuter: Boolean = true
         override def convertToGpu(): GpuExpression = GpuPosExplode(childExprs.head.convertToGpu())
       }),
+    expr[ReplicateRows](
+      "Given an input row replicates the row N times",
+      ExprChecks.projectOnly(
+        // The plan is optimized to run HashAggregate on the rows to be replicated. So this
```

Comment:
Please file a follow on issue for this. I don't understand why decimal128 is a problem for generate. Even if it is being fed into an aggregation we still support aggregating on decimal128 values. It is just a bit slower.  If these are all going to be used as keys to the hash aggregate, then it is a bit different, but that still needs a follow on issue to make it work.

Reply:
Sorry, decimal128 isn't the problem for generate here but all the `project` columns become keys for HashAggregateExec. So we end up running HashAggregate on CPU and then ReplicateRows on GPU in decimal128 case. 
I will update ReplicateRows to support  decimal 128(keeping the comment that grouping by 128-decimal is not supported yet).  
Could you please let me know if the follow on issue should track supporting grouping by 128-decimal in general OR it should be for ReplicateRows ?

I ran a simple query and the explain shows that HashAggregate is on CPU followed by ReplicateRows on GPU:

Dataframe:
+---+------------------------------+
| id|               amount           |
+---+------------------------------+
|100|100.3300000000000...|
|200|200.6600000000000...|
|300|300.9900000000000...|
+---+------------------------------+

Result of explain:
`testDf.select($"id", $"amount").exceptAll(testDf.select($"id", $"amount").filter("id > 100")).explain`

```
== Physical Plan ==
GpuColumnarToRow false
+- GpuProject [id#100, amount#101]
   +- GpuGenerate gpureplicaterows(sum#118L, id#100, amount#101), [id#100, amount#101], false, [id#100, amount#101]
      +- GpuCoalesceBatches targetsize(2147483647)
         +- GpuFilter (gpuisnotnull(sum#118L) AND (sum#118L > 0)), true
            +- GpuRowToColumnar targetsize(2147483647)
               +- *(2) HashAggregate(keys=[id#100, amount#101], functions=[sum(vcol#115L)])
                  +- Exchange hashpartitioning(id#100, amount#101, 200), ENSURE_REQUIREMENTS, [id=#511]
                     +- *(1) HashAggregate(keys=[id#100, amount#101], functions=[partial_sum(vcol#115L)])
                        +- GpuColumnarToRow false
                           +- GpuUnion
                              :- GpuRowToColumnar targetsize(2147483647)
                              :  +- LocalTableScan [vcol#115L, id#100, amount#101]
                              +- GpuRowToColumnar targetsize(2147483647)
                                 +- LocalTableScan [vcol#116L, id#111, amount#112]

```

Reply:
I still would like a follow on issue for decimal 128 so we are sure we don't miss it.